tests/slow/test_grpo_slow.py

```diff
@@ -188,3 +206,357 @@ def test_training_with_transformers_paged(self, model_name):
                 self.assertFalse(torch.equal(param, new_param), f"Parameter {n} has not changed.")
 
         release_memory(model, trainer)
+
+    @require_flash_attn
+    @require_bitsandbytes
+    @require_peft
+    @parameterized.expand(
+        [
+            ("HuggingFaceTB/SmolVLM-Instruct",),  # Only test the smaller model to avoid OOM
+        ]
+    )
+    def test_vlm_training(self, model_name):
+        """
+        Test VLM training with aggressive memory optimization.
+
+        This test uses multiple memory reduction techniques:
+        - 4-bit quantization with double quantization
+        - LoRA with very low rank (r=4)
+        - Minimal batch size (1) with gradient accumulation
+        - Small images (64x64 instead of 224x224)
+        - Short sequences (max_completion_length=8)
+        - Only 4 training samples
+        - Only 1 training step
+        - Gradient checkpointing and bfloat16
+        """
+
+        def data_gen(num_samples):
+            processor = AutoProcessor.from_pretrained(model_name)
```

Comment:
```suggestion
            processor = AutoProcessor.from_pretrained(model_name, use_fast=True, padding_side="left")
```

As the [docs](https://moon-ci-docs.huggingface.co/docs/trl/pr_3072/en/grpo_trainer#trl.GRPOTrainer.processing_class) say, `padding_side` should be `left`.